src/common/box_filters.c

```diff
@@ -63,14 +63,14 @@ static void blur_horizontal_1ch(float *const restrict buf, const int height, con
     const size_t index = (size_t)y * width;
     float *const restrict scanline = dt_get_perthread(scanlines,padded_size);
     // add up the left half of the window
-    for (int x = 0; x < radius && x < width ; x++)
+    for (int x = 0; x < MIN(radius,width) ; x++)
     {
       L += buf[index+x];
       hits++;
     }
     // process the blur up to the point where we start removing values
     int x;
-    for (x = 0; x <= radius && x < width; x++)
+    for (x = 0; x <= radius && x + radius < width; x++)
```

Comment:
Although the compiler is probably doing the right thing on interpreting this expression, it is still complicated to read and therefore would propose to add some braces:
```suggestion
    for (x = 0; x <= radius && ((x + radius) < width); x++)
```

Reply:
Agreed, yet PR is an important fix. I'll merge and do the clean-up in a separate commit.

Reply:
If you want I could also do that ;)

Reply:
Ah, I saw you already did that on master, alright.